experimental/cmd/mfrc522/main.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2016 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"periph.io/x/periph/conn/spi/spireg"
+	rf522 "periph.io/x/periph/experimental/devices/mfrc522"
+	"periph.io/x/periph/experimental/devices/mfrc522/commands"
+	"periph.io/x/periph/host"
+	"strconv"
+	"strings"
+)
+
+var (
+	sector = flag.Int("sector", 1, "Sector to access")
+	block  = flag.Int("block", 0, "Block to access")
+
+	rsPin  = flag.String("rs-pin", "13", "Reset pin")
+	irqPin = flag.String("irq-pin", "12", "IRQ pin")
+
+	keyCommand   = flag.Bool("wa", false, "Overwrite keys")
+	blockCommand = flag.Bool("wb", false, "Overwrite block by [0-15]")
+
+	spiDevice = flag.String("spidev", "/dev/spidev0.0", "SPI device")
+
+	key = flag.String("key", "", "Comma-separated key bytes")
+)
+
+func main() {
+
+	flag.Parse()
+
+	fmt.Println("Using pin settings: rsPin", *rsPin, "irqPin", *irqPin)
+
+	if _, err := host.Init(); err != nil {
+		log.Fatal(err)
+	}
+
+	currentAccessKey := rf522.DefaultKey
+
+	if *key != "" {
+		keyBytes := strings.SplitN(*key, ",", 6)
+		if len(keyBytes) != 6 {
+			log.Fatal("Key should consist of 6 decimal numbers")
```

Comment:
return an error, with mainImpl() flow.

Reply:
what is `mainImpl` flow?

Reply:
Here's one of the simplest example: https://github.com/google/periph/blob/master/cmd/ir/main.go

This permits thinking in term of error, not exceptions and this makes the code easier to copy-paste for users.